agent/engine/task_manager.go

```diff
@@ -267,6 +274,13 @@ func (mtask *managedTask) waitSteady() {
 	}
 }
 
+// steadyState returns if the task is in a steady state. Steady state is when task's desired
+// and known status are both RUNNING
+func (mtask *managedTask) steadyState() bool {
```

Comment:
nit, and possibly my java-brain speak here: can this be named "isSteadyState" or "isInSteadyState"?

Reply:
Hmm, don't have strong opinions here. `if task.steadyState()` conveys the 'is task in steady state?'  meaning in my brain. Golang prefers less verbosity in names (though we don't always stick to that convention). If you feel strongly about it, I can change it.